### DISCUSSION TITLE
Split string on null byte

### DISCUSSION THREAD

**Discussion by J-Cake** [nushell/nushell#10340] [Q&A] [answered]
Heyy, I'm trying to look at the `/proc/x/environ` file, which contains all environment vars, concatenated with null-bytes. I'm struggling to parse this file, as NuShell doesn't seem to support the `\0` notation for null bytes.

I've tried:
```nu
> open /proc/220595/environ | split row \0
> open /proc/220595/environ | split row '\0'
> open /proc/220595/environ | split row -r '\\0'
> open /proc/220595/environ | split row -r '\0'
> open /proc/220595/environ | split row \u0000
> open /proc/220595/environ | split row \x0
> open /proc/220595/environ | split row -r '\\x0'
> open /proc/220595/environ | split row ('00' | decode hex)
```

Of course I can do this easily with `grep` but that's beside the point. I'm hoping someone can enlighten me on how to achieve this

**Comment by amtoine** [comment, marked as answer]
i have this synthetic `foo` string, null separated
```nushell
> $foo | into binary
Length: 11 (0xb) bytes | printable whitespace ascii_other non_ascii
00000000:   66 6f 6f 00  62 61 72 00  62 61 7a                   foo0bar0baz
```
i can split it on `\0` with
```nushell
> $foo | split row (char --integer 0)
─┬───
0│foo
1│bar
2│baz
─┴───
```

> **Note**
> the variable has been generated with
> ```nushell
> let foo = ["foo", "bar", "baz"] | str join (char --integer 0)
> ```

**Reply by J-Cake** [reply]
Ah the `char --integer 0` is what I was looking for. Thank u sir

**Reply by amtoine** [reply]
happy to help :wink:
